### ISSUE TITLE
Allow penalty for Stan models

### ISSUE THREAD

**Issue by jaredlander** [tidymodels/parsnip#197]
The `penalty` argument in `linear_reg()` and `logistic_reg()` only applies to `glmnet`, `keras` and spark`. I think this could be added to `stan` with a slight change. This code shows the setup for `stan` and the `defaults` does not specify a `prior` which means it falls back to the default in `stan_glm()` which is a `normal(0, 2.5)` for predictors and `normal(0, 10)` for the intercept.

```
set_fit(
  model = "linear_reg",
  eng = "stan",
  mode = "regression",
  value = list(
    interface = "formula",
    protect = c("formula", "data", "weights"),
    func = c(pkg = "rstanarm", fun = "stan_glm"),
    defaults = list(family = expr(stats::gaussian))
  )
)
```

If a user specifies a `penalty`, `prior` can be set to `lasso(0, penalty)` or even `lasso(df, 0 , scale)`. 

The `mixture` argument is a bit trickier, but I believe a ridge prior would be `normal(0, diag(ncol(input_matrix))/ncol(input_matrix))` so maybe we can do a mixture of this and `laplace()` or `lasso()`.

**Comment by alexpghayes** [comment]
Just a quick note that the [Bayesian LASSO is bad](https://statmodeling.stat.columbia.edu/2017/11/02/king-must-die/) and people probably shouldn't use it. MAP estimates for a linear model with a laplace prior correspond to the frequentist LASSO, but Stan is going to return posterior means, which won't be very good.

**Comment by jaredlander** [comment]
You're right, it's not ideal. But that post even links to another where Andrew explains how to do a lasso in Stan, even if just in a "yes, we can do it, but maybe you shouldn't way." Same thing could be said here, it is technically possible.

That said, instead of the lasso prior the horseshoe via `hs()` would be an improvement over nothing and that is well endorsed by the Stan team.

**Comment by alexpghayes** [comment]
+1 for horseshoe, that'd be awesome!

**Comment by jaredlander** [comment]
It can he done with an `if` statement similar to how keras is handled. Have an `if` statement check if we want a penalty, then if so, add `prior=hs(...)` to the list of arguments passed to `stan_glm()`.
